app/Core/Src/API_simpleShapes.cpp

```diff
@@ -0,0 +1,259 @@
+/**
+ * @note Amount of points draw while drawing a circle
+ */
+#define MAX_CIRCLE_POINTS 100
+
+static int (*loggerCallback)(const char* message, int length) = nullptr;
+static void log_message(std::string message);
+static void API_setPixel(int, int, uint8_t);
+
+/**
+ * @brief Drawing line
+ *
+ * @param x_12 X coordinates of the points
+ * @param y_12 Y coordinates of the points
+ * @param color Color of the line
+ * @param weight Thickness of the corresponding line
+ * @param reserved Currently unused. Reserved for future use.
+ *
+ * @return -1 if error occured, otherwise 0 (Also a call to Logger callback is sent if ptr isn't NULL)
+ */
+int API_draw_line(int x0, int y0, int x1, int y1, int color, int weight, int reserved)
+{
+    (void)reserved;
+
+    if (OUTSIDE(x0, 0, VGA_DISPLAY_X) || OUTSIDE(x1, 0, VGA_DISPLAY_X)
+        || OUTSIDE(y0, 0, VGA_DISPLAY_Y) || OUTSIDE(y1, 0, VGA_DISPLAY_Y))
+    {
+        log_message("Line outside range of display");
+        return -1;
+    }
+
+    for (int i = 1; i < weight; i++)
+    {
+        API_draw_circle(x0, y0, i, color, 0);
+    }
+
+    int16_t steep = abs(y1 - y0) > abs(x1 - x0);
+    if (steep)
+    {
+        int temp = x0;
+        x0 = y0;
+        y0 = x0;
+        temp = x1;
+        x1 = y1;
+        y1 = temp;
+    }
+
+    if (x0 > x1)
+    {
+        int temp = x0;
+        x0 = x1;
+        x1 = temp;
+        temp = y0;
+        y0 = y1;
+        y1 = temp;
+    }
+
+    int16_t dx, dy;
+    dx = x1 - x0;
+    dy = abs(y1 - y0);
+
+    int16_t err = dx / 2;
+    int16_t ystep;
+
+    if (y0 < y1)
+    {
+        ystep = 1;
+    }
+    else
+    {
+        ystep = -1;
+    }
+
+    for (; x0 <= x1; x0++)
+    {
+        if (steep)
+        {
+            API_draw_circle(y0, x0, weight, color, 0);
+        }
+        else
+        {
+            API_draw_circle(x0, y0, weight, color, 0);
+        }
+        err -= dy;
+        if (err < 0)
+        {
+            y0 += ystep;
+            err += dx;
+        }
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Drawing circle
+ *
+ * @param x_y position on screen
+ * @param radius Radius of the circle
+ * @param color Color of the circle
+ * @param reserved Currently unused. Reserved for future use.
+ *
+ * @return -1 if error occured, otherwise 0
+ */
+int API_draw_circle(int x, int y, int radius, int color, int reserved)
+{
+    (void)reserved;
+
+    if (OUTSIDE(x, 0, VGA_DISPLAY_X) || OUTSIDE(x, 0, VGA_DISPLAY_Y))
+    {
+        log_message("X or Y is outside the allowed range");
+        return -1;
+    }
+
+    if (OUTSIDE(color, 0, MAX_COLOR_DEPTH))
+    {
+        log_message("Color outside of allowed range. Color scaling isn't supported.");
+        return -1;
+    }
+
+    if (OUTSIDE(radius, 0, MAX_RADIUS_SIZE))
+    {
+        log_message("Radius is outside of allowed range");
+        return -1;
+    }
+
+    for (int i = 0; i < MAX_CIRCLE_POINTS; i++)
+    {
+        float angle = i * 2 * (PI / 100);
```

Comment:
enable the FPU, maybe easier if it is done in this commit
`SCB->CPACR |= ((3UL << 10 * 2) | (3UL << 11 * 2));`